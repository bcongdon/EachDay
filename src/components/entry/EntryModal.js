import React, { Component } from 'react'
import { Modal } from 'semantic-ui-react'
import { PropTypes } from 'prop-types'
import EntryForm from './EntryForm'
import { openEntryModal, closeEntryModal } from '../../actions'
import { connect } from 'react-redux'

class EntryModal extends Component {
  static propTypes = {
    open: PropTypes.bool.isRequired,
    initialModalValues: PropTypes.object,
    closeEntryModal: PropTypes.func,
    trigger: PropTypes.element
  }

  constructor (props) {
    super(props)
    this.handleClose = this.handleClose.bind(this)
  }

  handleClose () {
    this.props.closeEntryModal()
  }

  render () {
    // Test for null/undefined or empty object
    const isNewEntry = !this.props.initialModalValues || Object.keys(this.props.initialModalValues).length === 0

    const actionText = isNewEntry ? 'Edit' : 'Create'
    return (
      <Modal open={this.props.open} onClose={this.handleClose} trigger={this.props.trigger}>
        <Modal.Header>{actionText} an Entry</Modal.Header>
        <Modal.Content>
          <EntryForm
            initialValues={this.props.initialModalValues}
            isNewEntry={isNewEntry} />
        </Modal.Content>
      </Modal>
    )
  }
}

function mapStateToProps (state) {
  return {
    open: state.entry.entryModalOpen,
    initialModalValues: state.entry.initialModalValues
  }
}

export default connect(mapStateToProps, { openEntryModal, closeEntryModal })(EntryModal)
