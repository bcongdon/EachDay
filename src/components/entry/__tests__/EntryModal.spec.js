import React from 'react'
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'
import { EntryModal } from '../EntryModal'
import { Modal } from 'semantic-ui-react'

jest.mock('universal-cookie')

const closeEntryModal = jest.fn()
const trigger = (<div />)

describe('entry component', () => {
  beforeEach(() => {
    jest.resetAllMocks()
  })

  it('should display correct text for new entry creation', () => {
    var modal = shallow(
      <EntryModal
        open
        initialModalValues={null}
        trigger={trigger}
        />
    )
    expect(modal.find(Modal.Header).props().content).toBe('Create an Entry')
    expect(toJson(modal)).toMatchSnapshot()
  })

  it('should display correct text for editing an entry', () => {
    var modal = shallow(
      <EntryModal
        open
        initialModalValues={{ rating: 5, notes: 'foo', date: '2017-01-01' }}
        trigger={trigger}
        />
    )
    expect(modal.find(Modal.Header).props().content).toBe('Edit an Entry')
    expect(toJson(modal)).toMatchSnapshot()
  })

  it('should close correctly', () => {
    var modal = shallow(
      <EntryModal
        open
        initialModalValues={{ rating: 5, notes: 'foo', date: '2017-01-01' }}
        closeEntryModal={closeEntryModal}
        trigger={trigger}
        />
    )
    modal.find(Modal).props().onClose()
    expect(closeEntryModal).toBeCalled()
  })
})
