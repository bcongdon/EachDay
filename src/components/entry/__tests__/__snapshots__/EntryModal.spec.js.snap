// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`entry component should display correct text for editing an entry 1`] = `
<Modal
  closeOnDimmerClick={true}
  closeOnDocumentClick={false}
  dimmer={true}
  onClose={[Function]}
  open={true}
  trigger={<div />}
>
  <ModalHeader
    content="Edit an Entry"
  />
  <ModalContent>
    <Connect(ReduxForm)
      initialValues={
        Object {
          "date": "2017-01-01",
          "notes": "foo",
          "rating": 5,
        }
      }
      isNewEntry={false}
    />
  </ModalContent>
</Modal>
`;

exports[`entry component should display correct text for new entry creation 1`] = `
<Modal
  closeOnDimmerClick={true}
  closeOnDocumentClick={false}
  dimmer={true}
  onClose={[Function]}
  open={true}
  trigger={<div />}
>
  <ModalHeader
    content="Create an Entry"
  />
  <ModalContent>
    <Connect(ReduxForm)
      initialValues={null}
      isNewEntry={true}
    />
  </ModalContent>
</Modal>
`;
